### PR TITLE
Solved BinarySearch

### PR DESCRIPTION
**Pull request by UreaLaden**
- [x] Top-level README “Table of Contents” is updated
 - [x] Feature tasks for this challenge are completed
 - [x] Unit tests written and passing
     - [x] “Happy Path” - Expected outcome
     - [x] Expected failure
     - [x] Edge Case (if applicable/obvious)
 - [x] README for this challenge is complete
     - [x] Summary, Description, Approach & Efficiency, Solution
     - [x] Link to code
     - [x] Picture of whiteboard

### CODE REVIEW

File: code-challenges/javaChallenges/lib/src/test/java/javaChallenges/BinarySearchTest.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package javaChallenges;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class BinarySearchTest {
```
Comment (Mattpet26) [inline code-challenges/javaChallenges/lib/src/test/java/javaChallenges/BinarySearchTest.java:7]: Great work on the tests, just don't forget to check for the base case. What if you were given an empty array?
